### PR TITLE
add outbound mqtt bindings

### PR DESCRIPTION
**Pull request by karthik2804**

### CODE REVIEW

File: examples/spin-outbound-mqtt/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+source venv/bin/activate
+```
+
+Install the required packages specified in the `requirements.txt` using the command:
+
+```bash
+pip3 install -r requirements.txt
+```
+
+For this example, a MQTT broker must be accessible at `127.0.0.1` with a username `user` and password `password should exist.
```
Comment (dicej) [inline examples/spin-outbound-mqtt/README.md:20]:
```suggestion
For this example, a MQTT broker must be accessible at `127.0.0.1`. A username `user` with password `password` should exist. 
```

File: src/spin_sdk/mqtt.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""Module for utilizing Spin Outbound MQTT"""
+
+from enum import Enum
+from spin_sdk.wit.imports.mqtt import Connection, Qos as Qos 
+
+def open(address: str, username: str, password: str, keep_alive_interval_in_secs: int) -> Connection:
+    """
+    Open a connection to the Mqtt instance at `address`.
+    
+    A `spin_sdk.wit.types.Err(spin_sdk.wit.imports.redis.ErrorInvalidAddress)` will be raised if the connection string is invalid.
```
Comment (dicej) [inline src/spin_sdk/mqtt.py:10]: Why is the `redis` module showing up here?

File: src/spin_sdk/mqtt.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    A `spin_sdk.wit.types.Err(spin_sdk.wit.imports.redis.ErrorTooManyConnection)` will be raised if there are too many open connections. Closing one or more previously opened connection using the `__exit__` method might help.
```
Comment (dicej) [inline src/spin_sdk/mqtt.py:12]:
```suggestion
    A `spin_sdk.wit.types.Err(spin_sdk.wit.imports.redis.ErrorTooManyConnections)` will be raised if there are too many open connections. Closing one or more previously opened connection using the `__exit__` method might help.
```

`pdoc` should be printing warnings when names like this are misspelled.  Please try running scripts/generate_docs.py again and pay close attention to any warnings.

File: src/spin_sdk/mqtt.py (hunk 2 above)
Comment (karthik2804) [reply src/spin_sdk/mqtt.py:10]: Copy paste without paying attention. Will fix.

**Review by dicej — APPROVED** [review]

**Pull request merged**